OS/DiskOS/terminal.lua

```diff
@@ -432,4 +477,60 @@ function term.loop() --Enter the while loop of the terminal
   end
 end
 
+--Given an input and a list of commands, return a list of strings that can be
+--used as autocomplete suggestion. For example if passed "c" and {"cls, copy"}
+--as parameters, will return {"ls", "opy"}.
+function term.autocomplete(input, commands)
+
+  result = {}
+  for k, f in ipairs(commands) do
+    if f:sub(1, input:len()) == input and input:len() < f:len() then
+      table.insert(result, f:sub(input:len()+1))
+    end
+  end
+
+  return result
+end
+
+--Display given suggestion on the command line
+function term.displaySuggestion(suggestion)
+  local cx, cy = printCursor()
+  local fw, fh = fontSize()
+  local x, y = 1+(fw+1)*cx, 1+(fh+1)*cy
+  rect(x-1,y-1,#suggestion*(fw+1)+1,fh+2,false,5)
+  print(suggestion,x,y)
+end
+
+--Clear given suggestion from the command line
+function term.clearSuggestion(suggestion)
+  local cx, cy = printCursor()
+  local fw, fh = fontSize()
```

Comment:
I see `fontSize()` is called inside of a function, The font size is static in current LIKO-12 versions, so it's more wise to move to line to the to of the file (maybe near the `termSize()` one.